search/includes/classes/class-search.php

```diff
@@ -616,13 +616,37 @@ public static function rate_limit_ep_query_integration( $skip ) {
 		if ( self::query_count_incr() > self::$max_query_count ) {
 			// Should be roughly half over time
 			if ( self::$query_db_fallback_value >= rand( 1, 10 ) ) {
+				self::record_ratelimited_query_stat();
 				return true;
 			}
 		}
 
 		return false;
 	}
 
+	public static function record_ratelimited_query_stat() {
+		$indexable = \ElasticPress\Indexables::factory()->get( 'post' );
+
+		if ( ! $indexable ) {
+			return;
+		}
+
+		// Can't use $this in static context
+		$es = self::instance();
```

Comment:
This class is starting to get messy :) Not something worth refactoring right now, but worth thinking how we could more cleanly integrate stats tracking in the future.